Add SLURM and NQSV examples (#1)
ExaESM-WP4/SLURM-Singularity-bindings: README.md

> # SLURM-Singularity-bindings

Reviewer: As this is pretty general now, let's rename? (Maybe in a different PR?)

```suggestion
# Batch-Singularity-bindings
```

Author: I'd say we better choose: "Batch scheduler Singularity bindings"?

> Wrapper scripts that enable host-system SLURM commands inside the container.

Reviewer: ```suggestion
Wrapper scripts that enable host-system batch scheduler commands inside the container.
```

Should be working especially with Red Hat (e.g. CentOS, ...) and Debian (e.g. Ubuntu, ...) Linux flavours.
Tested for CentOS (7 and 8), Debian (8, 9 and 10) and Ubuntu (18.04 and 20.04) container images.

> Please note, currently only `sinfo`, `squeue`, `sbatch` and `scancel` are supported.

Reviewer: ```suggestion
Please note, that for SLURM, currently only `sinfo`, `squeue`, `sbatch` and `scancel` are supported.
```

> 

Reviewer: `````suggestion

Before being able to use the `bind_batch.sh`,  you'll have to enable the configuration for your system. Do this with (e.g for Juwels)
```shell
ln -sf juwels.conf batch-bindings.conf
```

`````

```shell
$ singularity pull docker://centos:8

> $ ./bind_slurm.sh singularity shell --cleanenv centos_8.sif

Reviewer: Rename script to `bind_batch.sh`?
```suggestion
$ ./bind_batch.sh singularity shell --cleanenv centos_8.sif
```

# Batch scheduler bindings for Singularity containers

> This repository provides a wrapper script that adds HPC system batch scheduler functionality to Singularity containers. It was designed with container image portability in mind. Therefore, it does not require any target-system specific adaption to a container's `Dockerfile` and instead mounts the required system users, executables, libraries and configuration files into the container's file system at startup (see e.g. the [compatibility tests](./test_image_compatibility) for which only default container base images were used).

Reviewer: mounts --> bind mounts?

Author: Yes, done.

| HPC system | Singularity | compatible | not compatible |
| ---------: | :-----: | ------------- | ----- |
| NESH<br>Redhat-7.5_3.10.0<br>PBS | v3.5.3 | Ubuntu: 12, 14, 16, 18, 20<br>CentOS: 7, 8<br>Debian: 8, 9, 10<br>Amazon Linux: 1, 2<br> | Ubuntu: 10<br>CentOS: 6<br>Debian: 6, 7<br>--- |
| JUWELS<br>CentOS-7.8_3.10.0<br>SLURM | v3.6.1 | Ubuntu: 12, 14, 16, 18, 20<br>CentOS: 7, 8<br>Debian: 8, 9, 10<br>Amazon Linux: 1, 2 | Ubuntu: 10<br>CentOS: 6<br>Debian: 6, 7<br>--- |
| HLRN-B<br>Redhat-7.8_3.10.0<br>SLURM | v3.5.3 | Ubuntu: 12, 14, 16, 18, 20<br>CentOS: 7, 8<br>Debian: 8, 9, 10<br>Amazon Linux: 1, 2 | Ubuntu: 10<br>CentOS: 6<br>Debian: 6, 7<br>--- |
| HLRN-G<br>Redhat-7.8_3.10.0<br>SLURM | v3.2.1 | ---<br>CentOS: 7, 8<br>---<br>Amazon Linux: 1, 2 | Ubuntu: 10, 12, 14, 16, 18, 20<br>CentOS: 6<br>Debian: 6, 7, 8, 9, 10<br>--- |
| Mistral<br>Redhat-6.10_2.6.32<br>SLURM | v3.6.1 | Ubuntu: 10, 12, 14, 16<br>CentOS: 6, 7<br>Debian: 6, 7, 8, 9<br>Amazon Linux: 1 | Ubuntu: 18, 20<br>CentOS: 8<br>Debian: 10<br>Amazon Linux: 2 |

> > Please note, for containers to be compatible with the wrapper script it might also be required that an application's executables and libraries are not installed in one of the file system locations that are specified in the HPC system-specific configuration files. Singularity bind mounting takes precedence over the container's original file system contents, which may result in a broken application when a container application is started with the wrapper. It should be noted that, in general, the portability of a container image can be improved considerably by installing target applications not in a default file system location, but in application-specific folders inside e.g. the container's root path (and then explicitly adding these to the `$PATH` environment variable).

Reviewer: Block-Quote? I'd wrap this in `*` or `_`. But this is more a matter of taste, I think.

Author: I removed the blockquote styling. It might be best to just have this as normal text.
